db/commitlog/commitlog.cc

```diff
@@ -422,6 +460,60 @@ class db::commitlog::segment_manager : public ::enable_shared_from_this<segment_
     seastar::semaphore _reserve_recalculation_guard;
 };
 
+future<> db::commitlog::segment_manager::named_file::open(open_flags flags, file_open_options opt) noexcept {
+    assert(!*this);
+    auto f = co_await open_file_dma(_name, flags, opt);
+    // note this is not wholly true, since "create" means "create or open, whatever"
+    // but for all usage in this file, "create" will mean "new" (followed by truncate)
+    // so this check is ok.
+    auto existing_size = (flags & open_flags::create) == open_flags{}
```

Comment:
We also have open_flags::truncate, which is spearate from open_flags::create although they are often used together. So shouldn't you have checked open_flags::truncate here?

Reply:
If this was a top-level type, yes. In this case we adapt to the usage within commitlog, where "create" always means new (in the control flow).

Reply:
But don't you also have open_flags::truncate in the same places, so can check that flag instead?
Anyway, I just thought that if you used open_flags::truncate, in addition to (hopefully) being more correct or general, you could get rid of that comment where you needed to apologize for using the wrong flag.

Reply:
Perhaps. I guess the idea (somewhere) was that iff a file actually existed, we would not truncate it, since doing zeroing etc (that we do) will be faster if disk blocks already are there, but we still define the file size by to where we allocate/truncate/zero-write. Still, a teeny file size read in these cases as well is not the end of the world.